### PR TITLE
Feature/nick e conventional prepbufr

### PR DESCRIPTION
**Pull request by nicholasesposito**
This adds the conventional surface bufr data converter (using the python api) to the GDASApp. Data is from ADPSFC, SFCSHP and ADPUPA at the surface.

The two files added are:
    ~/ush/ioda/bufr2ioda/bufr2ioda_conventional_ps_prepbufr.py
    ~/parm/ioda/bufr2ioda/bufr2ioda_conventional_ps_prepbufr.json

This is validated in GDASApp issue #649 

The PRs for acft_profiles_prepbufr and GPSRO bufr should be added before this (PR #712 and #708 ) so that run_bufr2ioda.py works

There are a huge number of numpy/masked array variable sets because they kept overwriting for unknown reasons throughout, and kept resulting in data further down the line getting messed up, especially fill_values. Fill_values for numpy and ma are different for reasons only the python developers know, but as of right now, the fill_Values are all good and show up as "_"

### CODE REVIEW

**Comment by nicholasesposito** [comment]
The PRs for acft_profiles_prepbufr and GPSRO bufr should be added before this (PR #712 and #708 ) so that run_bufr2ioda.py works

**Comment by nicholasesposito** [comment]
There are a huge number of numpy/masked array variable sets because they kept overwriting for unknown reasons throughout, and kept resulting in data further down the line getting messed up, especially fill_values. Fill_values for numpy and ma are different for reasons only the python developers know, but as of right now, the fill_Values are all good and show up as "_"

File: ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py [hunk 1]
```diff
@@ -0,0 +1,792 @@
+#!/usr/bin/env python3
+# (C) Copyright 2023 NOAA/NWS/NCEP/EMC
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+import sys
+sys.path.append('/work2/noaa/da/nesposito/ioda_bundle_readlc_20231012/build/lib/python3.7/')
```
Comment (CoryMartin-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py]: remove the hard coded paths

File: ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py [hunk 2]
```diff
@@ -0,0 +1,792 @@
...
+    logger.debug(f"Running time for making QuerySet: {running_time} seconds")
+
+    # ==============================================================
+    # Open the BUFR file and execute the QuerySet to get ResultSet
+    # Use the ResultSet returned to get numpy arrays of the data
+    # ==============================================================
+    start_time = time.time()
+
+    logger.debug(f"Executing QuerySet to get ResultSet ...")
+    with bufr.File(DATA_PATH) as f:
```
Comment (CoryMartin-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py:180]: Any reason why these need opened up 3x? Or is this a BUFR quirk?

Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py:180]: It's a quirk. From what I understand, it goes through the whole thing on the ADPSFC data query, and then at the end it has to rewind. I found it easier to open it again.

File: ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py (hunk 1 above)
Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py]: done

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by ADCollard — APPROVED** [review]

**Pull request merged**